### PR TITLE
fix(fmgc): make des spd lim available prior to descent

### PR DESCRIPTION
**Pull request by tracernz**
Fixes #[issue_no]

## Summary of Changes

- Make DES SPD LIM available on VERT REV for all waypoints with DES constraints, and when flight phase is DES (without VNAV we don't have ToD so this is a compromise)
- Fix a couple of copy-pasta bugs with DES SPD LIM

## Additional context

Discord username (if different from GitHub):

## Testing instructions

With a flightplan, but no ZFW/BLOCK entry (i.e. no predictions available):
Check that CLB SPD LIM can be altered by VERT REV on from airport and SID waypoints, check that DES SPD LIM can be altered by VERT REV on to airport and STAR/APPR waypoints, without affecting each other.
Check that no SPD LIM is shown on VERT REV for enroute waypoints (unless a constraint type has been assigned).
Enter ZFW/BLOCK to get predictions on INIT B:
Check that CLB SPD LIM can be altered by VERT REV on from airport and SID waypoints, check that DES SPD LIM can be altered by VERT REV on to airport and STAR/APPR waypoints, without affecting each other.

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page

### CODE REVIEW

File: flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/CDU/A320_Neo_CDU_VerticalRevisionPage.js [hunk 1]
```diff
@@ -157,16 +162,7 @@ class CDUVerticalRevisionPage {
 
                 const climb = mcdu.flightPhaseManager.phase < FmgcFlightPhases.CRUISE;
```
Comment (BlueberryKing) [inline flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/CDU/A320_Neo_CDU_VerticalRevisionPage.js:163]: This appears to be unused.

```suggestion
```

**Review by BlueberryKing — CHANGES_REQUESTED** [review]

**Review by BlueberryKing — APPROVED** [review]

**Comment by mico975** [comment]
Quality Assurance Tester Report

Discord          : mico#3145
Object of testing: #7529 
Tier of Testing  : 1
Date             : 05/10/2022

Testing Process: 
Check that CLB SPD LIM can be altered by VERT REV on from airport and SID waypoints - **OK**
Check that DES SPD LIM can be altered by VERT REV on to airport and STAR/APPR waypoints  - **OK**
Without affecting each other. - **OK**

Negatives:       
N/A

Testing Results: 
Passed

Conclusions: 
LGTM

**Comment by tracernz** [comment]
Prior to predictions being available SPD LIM is only available on the origin and destination... I will make some tweaks to this.

**Review by BlueberryKing — APPROVED** [review]
🚀

**Comment by pinatacolada** [comment]
Just a reminder that after this pr was opened,  the clb/des spd lim was correlated to the spd lim flag, that is included along the entire sid/star points

**Comment by tracernz** [comment]
> Just a reminder that after this pr was opened, the clb/des spd lim was correlated to the spd lim flag, that is included along the entire sid/star points

Yes, thanks again for the new data points. 👍

**Review by BlueberryKing — APPROVED** [review]

**Pull request merged**